spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/DeploymentMessageConsumer.java

```diff
@@ -79,46 +86,84 @@ public void consumeMessage(DeploymentMessage deploymentMessage) throws Exception
-	 * @param deploymentMessage the deployment message
+	 * @param message the deployment message
 	 */
 	@SuppressWarnings({"rawtypes", "unchecked"})
-	private void processDeploymentMessage(ResourceDeployer deployer, DeploymentMessage deploymentMessage) {
-		DeploymentAction deploymentAction = deploymentMessage.getDeploymentAction();
-		String name = deploymentMessage.getUnitName();
-		switch (deploymentAction) {
-			case create:
-			case createAndDeploy: {
-				if (deployer instanceof StreamDeployer) {
-					deployer.save(new StreamDefinition(name, deploymentMessage.getDefinition()));
-				}
-				else if (deployer instanceof JobDeployer) {
-					deployer.save(new JobDefinition(name, deploymentMessage.getDefinition()));
-				}
-				if (DeploymentAction.createAndDeploy.equals(deploymentAction)) {
-					deployer.deploy(name, Collections.<String, String>emptyMap());
+	private void processDeploymentMessage(ResourceDeployer deployer, DeploymentMessage message) {
+		DeploymentAction deploymentAction = message.getDeploymentAction();
+		String name = message.getUnitName();
+		String errorDesc = null;
+
+		try {
+			switch (deploymentAction) {
+				case create:
+				case createAndDeploy: {
+					if (deployer instanceof StreamDeployer) {
+						deployer.save(new StreamDefinition(name, message.getDefinition()));
+					}
+					else if (deployer instanceof JobDeployer) {
+						deployer.save(new JobDefinition(name, message.getDefinition()));
+					}
+					if (DeploymentAction.createAndDeploy.equals(deploymentAction)) {
+						deployer.deploy(name, Collections.<String, String>emptyMap());
+					}
+					break;
 				}
-				break;
+				case deploy:
+					deployer.deploy(name, message.getDeploymentProperties());
+					break;
+				case undeploy:
+					deployer.undeploy(name);
+					break;
+				case undeployAll:
+					deployer.undeployAll();
+					break;
+				case destroy:
+					deployer.delete(name);
+					break;
+				case destroyAll:
+					deployer.deleteAll();
+					break;
 			}
-			case deploy:
-				deployer.deploy(name, deploymentMessage.getDeploymentProperties());
-				break;
-			case undeploy:
-				deployer.undeploy(name);
-				break;
-			case undeployAll:
-				deployer.undeployAll();
-				break;
-			case destroy:
-				deployer.delete(name);
-				break;
-			case destroyAll:
-				deployer.deleteAll();
-				break;
+		}
+		catch (Throwable t) {
+			errorDesc = ZooKeeperUtils.getStackTrace(t);
+			throw t;
+		}
+		finally {
+			writeResponse(message, errorDesc);
+		}
+
+	}
+
+	/**
+	 * Write the deployment message response.
+	 *
+	 * @param message    deployment message
+	 * @param errorDesc  error description if an error occurred processing
+	 *                   the request
+	 */
+	private void writeResponse(DeploymentMessage message, String errorDesc) {
+		try {
+			String requestId = message.getRequestId();
+			if (StringUtils.hasText(requestId)) {
+				logger.info("Processed deployment request " + requestId);
```

Comment:
Agreed on making this debug; hopefully you kept the request id anyway - this can come in useful when debugging.